src/talk/renderer/notifications/notifications.store.js

```diff
@@ -11,13 +11,31 @@
  */
 
 import { Howl } from 'howler'
-import { getRequestToken } from '@nextcloud/auth'
+import { getCurrentUser, getRequestToken } from '@nextcloud/auth'
 import { listen } from '@nextcloud/notify_push'
 import { loadState } from '@nextcloud/initial-state'
-import { emit } from '@nextcloud/event-bus'
+import { emit, subscribe } from '@nextcloud/event-bus'
 import { generateFilePath } from '@nextcloud/router'
 import { getNotificationsData } from './notifications.service.js'
 import { appData } from '../../../app/AppData.js'
+import { setInitialState } from '../../../shared/initialState.service.js'
+
+/**
+ * Enables or disables the sound for notifications and calls in the initial state
+ * @param {boolean} playSound - Whether to enable play a sound
+ */
+const setPlaySoundState = (playSound) => {
+	setInitialState('notifications', 'sound_notification', playSound)
+	setInitialState('notifications', 'sound_talk', playSound)
+}
+
+// Listen for user status changes to disable sounds on DND and enable it back otherwise
+subscribe('user_status:status.updated', (userStatus) => {
```

Comment:
Does the status event trigger initially? Or what happens when you start the client while you're already on DND?

Reply:
Well, yes and no.

Initially, when it is restored from the cache — no.

In 1 second when it is requested from the server — yes.

Makes sense to also emit initially?

Reply:
well 1s is good enough. Just don't want to wait until it actually changed.

Reply:
Done anyway, it seems better to use the status initially. Like if GET user status request is slower than loading new notifications.